streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -1526,6 +1526,25 @@ abstract class ZStream[-R, +E, +O](
     }
   }
 
+  /**
+   * Intersperse stream with provided element similar to <code>List.mkString</code>.
+   */
+  final def intersperse[O1 >: O](middle: O1): ZStream[R, E, O1] =
```

Comment:
@fagossa Could we add a variant that also adds start/end markers? This could simply be `ZStream(start) ++ intersperse(middle) ++ ZStream(end)`.

Reply:
done!